sanic/testing.py

```diff
@@ -114,6 +115,9 @@ def _collect_request(request):
             url = "{scheme}://{host}:{port}{uri}".format(
                 scheme=scheme, host=host, port=port, uri=uri
             )
+        # Tests construct URLs using PORT = None, which means random port not
+        # known until this function is called, so fix that here
+        url = url.replace(":None/", f":{port}/")
```

Comment:
a little bit concern on this, probably should avoid this kinda of hacky pattern.

Reply:
Avoiding this would require major restructuring of `testing.py` which I am not quite up to, or rewriting/disabling all tests that depend on this.